src/en/mangaowlto/src/eu/kanade/tachiyomi/extension/en/mangaowlto/MangaOwlTo.kt

```diff
@@ -0,0 +1,227 @@
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/$collectionUrl?ordering=-view_count&page=$page", headers)
+    }
+
+    override fun popularMangaSelector() = "div.manga-item"
+
+    override fun popularMangaFromElement(element: Element): SManga {
+        val manga = SManga.create()
+        element.select("a:nth-child(2)").let {
+            manga.setUrlWithoutDomain(it.attr("href"))
+            manga.title = it.text()
+        }
+        manga.thumbnail_url = element.select("a:nth-child(1) > img").attr("src")
+
+        return manga
+    }
+
+    override fun popularMangaNextPageSelector() = "div.pagination > a[title='Next']"
+
+    // Latest
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/$collectionUrl?ordering=-modified_at&page=$page", headers)
+    }
+
+    override fun latestUpdatesSelector() = popularMangaSelector()
+
+    override fun latestUpdatesFromElement(element: Element): SManga = popularMangaFromElement(element)
+
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+
+    // Search
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        return if (query.isNotEmpty() || filters.isEmpty()) {
+            // Search won't work together with filter
+            GET("$baseUrl/$searchPath?q=$query&page=$page", headers)
+        } else {
+            val url = "$baseUrl/$collectionUrl".toHttpUrl().newBuilder()
+            filters.forEach { filter ->
+                when (filter) {
+                    is SortFilter -> if (!filter.toUriPart().isNullOrEmpty()) {
+                        url.addQueryParameter("ordering", filter.toUriPart())
+                    }
+                    is StatusFilter -> if (!filter.toUriPart().isNullOrEmpty()) {
+                        url.addQueryParameter("status", filter.toUriPart())
+                    }
+                    is GenreFilter ->
+                        filter.state
+                            .filter { it.state }
+                            .forEach { url.addQueryParameter("genres", it.uriPart) }
+                    else -> {}
+                }
+            }
+
+            url.apply {
+                addQueryParameter("page", page.toString())
+            }
+            GET(url.toString(), headers)
+        }
+    }
+
+    override fun searchMangaSelector() = popularMangaSelector()
+
+    override fun searchMangaFromElement(element: Element): SManga = popularMangaFromElement(element)
+
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+
+    // Manga summary page
+
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        val slug = manga.url.substringAfterLast("/")
+        return GET("$API/stories/$slug", headers)
+    }
+
+    override fun mangaDetailsParse(document: Document) =
+        json.decodeFromString<MangaOwlTitle>(document.body().text()).toSManga()
```

Comment:
```suggestion
    override fun mangaDetailsParse(response: Response) =
        json.decodeFromString<MangaOwlTitle>(response.body.string()).toSManga()
   
    override fun mangaDetailsParse(document: Document) = throw UnsupportedOperationException()
```

Either do this, or switch to inheriting `HttpSource` and use the API entirely.